gsadjust/pyqt_models.py

```diff
@@ -1724,22 +1734,16 @@ def setData(self, index, value, role):
 
         if role == QtCore.Qt.EditRole:
             if index.isValid() and 0 <= index.row():
-                if len(str(value)) > 0:
+                if value:
                     datum = self.datums[index.row()]
                     column = index.column()
-                    if column == DATUM_STATION:
-                        datum.station = str(value)
-                    elif column == DATUM_G:
-                        datum.g = float(value)
-                    elif column == DATUM_SD:
-                        datum.sd = float(value)
-                    elif column == DATUM_DATE:
-                        datum.date = value
-                    elif column == MEAS_HEIGHT:
-                        datum.meas_height = float(value)
-                    elif column == GRADIENT:
-                        datum.gradient = float(value)
+
+                    attr, vartype = _attrs.get(column, (None, None))
+                    if attr:
+                        setattr(self, attr, vartype(value))
```

Comment:
To perform the operation we now get the values from our self._attrs dictionary, returning the attribute name in `attr` and the type of the value in `vartype`.

If you want to enforce this typing more strongly, you can implement getters and setters for these attributes which perform this type conversion on set automatically.